cmd/ocm-backplane/console/console_test.go

```diff
@@ -294,4 +376,127 @@ var _ = Describe("console command", func() {
+			_, err4 := o.getContainerEngineImpl()
+			Expect(err4).To(MatchError(ContainSubstring("container engine can only be one of podman|docker")))
+			os.Setenv("BACKPLANE_DEFAULT_OPEN_BROWSER", "")
+		})
+	})
+
+	// Putting everything together, we could call the .run function to test the particular functionality
+	Context("Once we have validated container runtimes, we need to start the containers in which the console resides", func() {
+
+		It("Prior to running the console we need to create a cobra.command object and verify the flag entries are created", func() {
+			consoleCmd := NewConsoleCmd()
+			flags := consoleCmd.Flags()
+			Expect(reflect.TypeOf(flags) == reflect.TypeOf(&pflag.FlagSet{})).To(BeTrue())
+		})
+
+		It("The run function checks configurations and runs the container", func() {
+			// createPathPodman()
+			oldpath := createPathDocker()
+			// Create a new client set or deployment to be read
+			createClientSet = func(c *rest.Config) (kubernetes.Interface, error) {
+				return testclient.NewSimpleClientset(&appsv1.DeploymentList{Items: []appsv1.Deployment{{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: ConsoleNS,
+						Name:      ConsoleDeployment,
+					},
+					Spec: appsv1.DeploymentSpec{
+						Template: corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{{
+									Name:  "console",
+									Image: "testrepo.com/test/console:latest",
+								}},
+							},
+						},
+					},
+				}}}), nil
+			}
+
+			// Define a Cluster info, 4.13 since we're not testing monitoring in this scenario
+			clusterInfo, _ := cmv1.NewCluster().
+				CloudProvider(cmv1.NewCloudProvider().ID("aws")).
+				Product(cmv1.NewProduct().ID("dedicated")).
+				AdditionalTrustBundle("REDACTED").
+				Proxy(cmv1.NewProxy().HTTPProxy("http://my.proxy:80").HTTPSProxy("https://my.proxy:443")).
+				OpenshiftVersion("4.13.0").Build()
+
+			// Set Browser opening to false
+			os.Setenv("BACKPLANE_DEFAULT_OPEN_BROWSER", "FALSE")
+			setupConfig()
+
+			// Set some mock varibles,
+			mockOcmInterface.EXPECT().GetPullSecret().Return(pullSecret, nil).AnyTimes()
+			mockOcmInterface.EXPECT().GetClusterInfoByID(clusterID).Return(clusterInfo, nil).AnyTimes()
+			mockOcmInterface.EXPECT().GetOCMAccessToken().Return(&fakeToken, nil).AnyTimes()
+			// Create a new ocmEnvironment for mock use
+			ocmEnvironment, _ := cmv1.NewEnvironment().BackplaneURL("fakeBackPlaneUrl").Build()
+			// Tell mockOCM interface to return ocnEnvironment
+			mockOcmInterface.EXPECT().GetOCMEnvironment().Return(ocmEnvironment, nil).AnyTimes()
+			o := consoleOptions{}
+			o.port = "1337"
+			o.url = "http://127.0.0.2:1447"
+			o.openBrowser = false
+			o.containerEngineFlag = DOCKER
+
+			argvstrs1 := []string{"console_GO_test"}
+			err := o.run(&cobra.Command{}, argvstrs1)
+			// There isn't any way to stop the o.run loop while testing, so we have to feed it a varible to force a termination
```

Comment:
you may able to use `cancelContext` for the cmd execution 
pls refer : 
https://pkg.go.dev/github.com/spf13/cobra#Command.ExecuteContextC
https://blog.ksub.org/bytes/2019/10/07/using-context.context-with-cobra/

Reply:
I've implemented changes in a separate branch: 

https://github.com/rhs-harliu/backplane-cli/tree/harliu-console-tests-context

Alongside an alternative:
https://github.com/rhs-harliu/backplane-cli/tree/harliu-console-tests-interfaces

I'll discuss when I have time.